src/parser.ts

```diff
@@ -90,6 +91,24 @@ interface AttributeKeyMap {
 }
 
 export class GraphMLParser {
+    public parse(text: string, cb?: Function) {
+
+        let parser = new Parser();
+
+        parser.parseString(text, (err: any, document: schema.GraphMLDocument) => {
+            if (err) {
+                cb(err, []);
+                return;
+            }
+
+            let graphs = document.graphml.graph;
+            let keys = document.graphml.key;
+            cb(err, graphs.map((graph: schema.GraphElement) => new GraphBuilder().build(keys, graph)));
+        });
+    }
+}
+
+class GraphBuilder {
```

Comment:
I renamed `GraphMLParser` to `GraphBuilder` and this class is used by new `GraphMLParser`. The reason of this change is that we should parse a graph multiple times.

Reply:
This all looks good! The only thing is that generally the putting "Builder" on the end of a name suggests that the builder pattern is being used, which isn't  really the case here. Is have a build method really necessary anymore? Maybe we should rename `GraphBuilder` to something like `Graph` or `ParsedGraph` and pass the keys and graph directly to the constructor?

Reply:
Good point. 

There are some options like:

1. remove `GraphBuilder` and move all functions to `Graph` class. Then add `Graph.create(GraphKeyElement[], GraphElement)` static function, which is a factory method for `Graph`.
2. rename `GraphBuilder` to `ParsedGraph` and pass the keys and graph directly to the constructor (you mentioned above). 

I think 1 is simple enough in this case.